modules/core/src/v2/coins/stx.ts

```diff
@@ -87,9 +89,20 @@ export class Stx extends BaseCoin {
     // TODO: Implement when available on the SDK.
     return Bluebird.resolve(true).asCallback(callback);
   }
+
   verifyAddress(params: VerifyAddressOptions): boolean {
-    // TODO: Implement when available on the SDK.
-    throw true;
+    const { address } = params;
+    if (!this.isValidAddress(address)) throw new InvalidAddressError(`invalid address: ${address}`);
+
+    try {
+      const [version] = c32addressDecode(address);
+      const versionString = accountLib.Stx.AddressVersion[version];
+      if (versionString === undefined) throw new InvalidAddressError(`invalid address version: ${address}`);
```

Comment:
Can we create an enum of valid address versions and ask if versionString is in the enum?